prov/sharp: sharp provider can be mocked

ldorau opened the pull request:
In order to mock SHARP functions run:
```sh
$ ./configure --enable-sharp=mocked
```
or
```sh
$ ./configure --enable-sharp=auto
```

Running:
```sh
$ ./configure --enable-sharp=yes
```
make use of SHARP libraries instead of mocks.

Signed-off-by: Lukasz Dorau <lukasz.dorau@intel.com>

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/grom72/libfabric/3)
<!-- Reviewable:end -->

grom72 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/grom72/libfabric/3#-:-NJo2VUXArx_m9YMaiez:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @ldorau)



<!-- Sent from Reviewable.io -->

ldorau committed:
prov/sharp: sharp provider can be mocked

In order to mock SHARP functions run:
$ ./configure --enable-sharp=mocked
or
$ ./configure --enable-sharp=auto

Running:
$ ./configure --enable-sharp=yes
make use of SHARP libraries instead of mocks.

Signed-off-by: Lukasz Dorau <lukasz.dorau@intel.com>

grom72 reviewed (approved):
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/grom72/libfabric/3)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @ldorau)



<!-- Sent from Reviewable.io -->
